### ISSUE TITLE
Checkbox stories have accessibility violations

### ISSUE THREAD

**Issue by snowystinger** [Frontify/fondue#960]
Hey! Saw you were using react-aria/stately. Awesome and thank you so much. I hope you don't mind, but I was perusing the components you've built and noticed that the Checkbox has a few accessibility violations in the storybook.
https://fondue-components.frontify.com/?path=/story/components-checkbox--checkbox

As well as a few that AXE didn't pick up on
No buttons inside a label - https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label#interactive_content
No nesting of labels - https://www.w3.org/TR/2013/CR-html5-20130806/forms.html#the-label-element (see "[Phrasing content](https://www.w3.org/TR/2013/CR-html5-20130806/dom.html#phrasing-content-1), but with no descendant [labelable elements](https://www.w3.org/TR/2013/CR-html5-20130806/forms.html#category-label) unless it is the element's [labeled control](https://www.w3.org/TR/2013/CR-html5-20130806/forms.html#labeled-control), and no descendant [label](https://www.w3.org/TR/2013/CR-html5-20130806/forms.html#the-label-element) elements.)

**Comment by ryancarville** [comment]
Hey @snowystinger! 👋 

Thank you for the feedback!  We are currently re-working our `checkbox` component and will be sure to keep these accessibility violation in mind.  I will close this issue now, but please let us know if you see anything else that will help us make it a great UX for all our users!
